driver-async/src/main/com/mongodb/async/client/MongoClients.java

```diff
@@ -218,25 +224,57 @@ public static CodecRegistry getDefaultCodecRegistry() {
         return com.mongodb.MongoClientSettings.getDefaultCodecRegistry();
     }
 
-    @SuppressWarnings("deprecation")
-    private static StreamFactory getStreamFactory(final MongoClientSettings settings, final String streamType,
-                                                  final boolean isHeartbeat) {
+
+    private static MongoClient createWithTlsChannel(final MongoClientSettings settings,
+                                                    @Nullable final MongoDriverInformation mongoDriverInformation) {
+        if (!isJava8()) {
+            throw new MongoClientException("TLS is only supported natively with Java 8 and above. Please use Netty instead");
+        }
+        final TlsChannelStreamFactoryFactory streamFactoryFactory = new TlsChannelStreamFactoryFactory();
+        StreamFactory streamFactory = streamFactoryFactory.create(settings.getSocketSettings(), settings.getSslSettings());
+        StreamFactory heartbeatStreamFactory = streamFactoryFactory.create(settings.getHeartbeatSocketSettings(),
+                settings.getSslSettings());
+        return createMongoClient(settings, mongoDriverInformation, streamFactory, heartbeatStreamFactory,
+                new Closeable() {
+                    @Override
+                    public void close() {
+                        streamFactoryFactory.close();
+                    }
+                });
+    }
+
+    private static boolean isJava8() {
+        try {
+            Class.forName("java.time.Instant");
```

Comment:
Is there a more reliable way?  It's what we do elsewhere.

Reply:
Its good enough :+1: